OCPBUGS-76497: Added note to nw-ingress-gateway-api-manage-succession… (#110130)
openshift/openshift-docs: modules/nw-ingress-gateway-api-manage-succession.adoc

Starting in {product-title} 4.19, the Ingress Operator manages the lifecycle of any Gateway API custom resource definitions (CRDs). This means you cannot create, update, or delete any CRDs within the API groups grouped under Gateway API.

Updating from a version before {product-title} 4.19 where the API CRD management was not present requires replacing or removing any Gateway API CRDs. 

[WARNING]
====
Updating or deleting Gateway API resources can result in downtime and loss of service or data. Be sure you understand how this impacts your cluster before performing the steps in this procedure. If necessary, back up any Gateway API objects in YAML format in order to restore it later.
====

The updated CRDs must conform to the specific {product-title} specification that is required by the Ingress Operator. {product-title} version 4.19 requires Gateway API Standard version 1.2.1 CRDs.

[IMPORTANT]
====

> When upgrading your {product-title} 4.18 cluster to {product-title} 4.19, you might see a `Gateway API CRDs have been detected` message. A cluster administrator must understand the implications before applying the `ack-4.18-gateway-api-management-in-4.19` acknowledgment. {product-title} 4.19 can then manage the Gateway API CRDs during the upgrade operation.

Reviewer: This note looks good.  However, it would be helpful to have an explicit procedure with the command:
```shell
oc -n openshift-config patch configmap admin-acks --patch '{"data":{"ack-4.18-gateway-api-management-in-4.19":"true"}}'
```

Author: ✔️

[id="nw-ingress-gateway-api-manage-succession_{context}"]
= Preparing for Gateway API management succession by the Ingress Operator

[role="_abstract"]
To ensure your Gateway API custom resource definitions (CRDs) conform to required {product-title} specifications, allow the Ingress Operator to manage their lifecycle. Preparing your older {product-title} environment for this automated management helps prevent resource conflicts and deployment errors.

> Starting in {product-title} 4.19, the Ingress Operator manages the lifecycle of any Gateway API custom resource definitions (CRDs). This means you cannot create, update, or delete any CRDs within the API groups grouped under Gateway API.

Reviewer 2: groups grouped??
```suggestion
Starting in {product-title} 4.19, the Ingress Operator manages the lifecycle of any Gateway API custom resource definitions (CRDs). This means you cannot create, update, or delete any CRDs within the API groups under the Gateway API.
```

> Updating from a version before {product-title} 4.19 where the API CRD management was not present requires replacing or removing any Gateway API CRDs. 

Reviewer 2: [ISG suggests](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#l) using _earlier_.

```suggestion
Updating from a version earlier than {product-title} 4.19 where the API CRD management was not present requires replacing or removing any Gateway API CRDs. 
```

[WARNING]
====
Updating or deleting Gateway API resources can result in downtime and loss of service or data. Be sure you understand how this impacts your cluster before performing the steps in this procedure. If necessary, back up any Gateway API objects in YAML format in order to restore it later.
====

The updated CRDs must conform to the specific {product-title} specification that is required by the Ingress Operator. {product-title} version 4.19 requires Gateway API Standard version 1.2.1 CRDs.

[IMPORTANT]
====
When upgrading your {product-title} 4.18 cluster to {product-title} 4.19, you might see a `Gateway API CRDs have been detected` message. A cluster administrator must understand the implications before applying the `ack-4.18-gateway-api-management-in-4.19` acknowledgment. {product-title} 4.19 can then manage the Gateway API CRDs during the upgrade operation.

You can run the following command to apply the `ack-4.18-gateway-api-management-in-4.19` acknowledgment:

[source,terminal]
----

> $ oc -n openshift-config patch configmap admin-acks --patch '{"data":{"ack-4.18-gateway-api-management-in-4.19":"true"}}'

Reviewer: The command is missing the `--type=merge` option:
```suggestion
$ oc -n openshift-config patch configmap admin-acks --patch '{"data":{"ack-4.18-gateway-api-management-in-4.19":"true"}}' --type=merge
```
I haven't tried the command without `--type=merge`.

Author: :heavy_check_mark:
